search/search_service/api/search.py

```diff
@@ -33,20 +35,28 @@ def post(self) -> Iterable[Any]:
         request_data = SearchRequestSchema().loads(json.dumps(request.get_json()))
 
         resources: List[AmundsenResource] = []
-        for r in request.get_json().get('resource_types'):
+        for r in request_data.resource_types:
             resource = RESOURCE_STR_MAPPING.get(r)
             if resource:
                 resources.append(resource)
             else:
                 err_msg = f'Search for invalid resource "{r}" requested'
                 return {'message': err_msg}, HTTPStatus.BAD_REQUEST
 
+        highlight_options: Dict[Resource, HighlightOptions] = {}
+        request_options = request_data.highlight_options
+        if request_options:
+            for r in request_options.keys():
```

Comment:
the serdes at top on line 35 should have made all this adhoc stuff unnecessary?

Reply:
the loads doesn't seem to work recursively so while the request is a SearchRequest object after line 35 highlight options is a dict

Reply:
still don't like this